overwatch/src/services/resources.rs

```diff
@@ -123,31 +122,32 @@ where
         &self.operator_fuse_sender
     }
 
-    /// Retrieves the inbound relay consumer from the channel.
+    /// Retrieves the [`InboundRelay`]'s receiver from the channel and rebuilds
+    /// a new [`InboundRelay`].
     ///
-    /// Only one inbound relay exists at a time.
+    /// Only one [`InboundRelay`] exists at a time for a given `Service`.
     ///
     /// This function must be called only if awaiting a `Service` to be Dropped,
-    /// which is when the inbound relay consumer is returned.
+    /// which is when the [`InboundRelay`]'s receiver is returned.
     ///
     /// # Errors
     ///
-    /// If the inbound relay already exists in [`ServiceResources`].
+    /// If the [`InboundRelay`] already exists in [`ServiceResources`].
     ///
     /// # Panics
     ///
-    /// If the consumer cannot be retrieved from the channel.
-    pub fn retrieve_inbound_relay_consumer(&mut self) -> Result<(), String> {
+    /// If the [`InboundRelay`]'s receiver cannot be retrieved from the channel.
+    pub fn rebuild_inbound_relay(&mut self) -> Result<(), String> {
```

Comment:
I guess there's no way to expose this function as well as the `as_handle` only in case the inbound relay is already set for it?

Reply:
I think actually we could just panic here. Meaning something went really wrong and the service wasn't droped when it was supposed to.